vue/mixins/locales.js

```diff
@@ -0,0 +1,156 @@
+         * Sets the fallback locale.
+         *
+         * @param {String} locale  The ISO-15897 standard locale definition string to be set.
+         */
+        setLocaleFallback(locale) {
+            this.$store.commit("SET_LOCALE_FALLBACK", locale);
+        },
+        /**
+         * Gets the fallback locale.
+         *
+         * @returns {String} The fallback locale.
+         */
+        getLocaleFallback() {
+            return this.$store.state.localePlugin.localeFallback;
+        },
+        /**
+         * Set the locales information.
+         *
+         * @param {Object} locales  The locales information object. Example:
+         * { en_us: { "ripe_twitch.value_a.value_b": "Value Example" } }.
+         */
+        setLocales(locales = {}) {
+            this.$store.commit("SET_LOCALES", locales);
+        },
+        /**
+         * Gets the object with all the locales.
+         *
+         * @returns {Object} The object with all the locales.
+         */
+        getLocales() {
+            return this.$store.state.localePlugin.locales;
+        },
+        /**
+         * Appends new locales information to the specified locale. If the locale isn't
+         * specified, it will use the already chosen locale.
+         *
+         * @param {Object} locales  The locales information object. Example:
+         * { en_us: { "ripe_twitch.value_a.value_b": "Value Example" } }.
+         * @param {String} locale  The ISO-15897 standard locale definition string to be set.
+         */
+        addLocales(locales = {}, locale = null) {
+            locale = locale || this.getLocale();
+
+            const localeObj = this.getLocales()[locale] || {};
+            const updatedLocaleObj = { ...localeObj, ...locales };
+            this.setLocales({
+                ...this.getLocales(),
+                ...{ [locale]: updatedLocaleObj }
+            });
+        },
+        /**
+         * Should be called when first opening the extension. It setups the locales plugin by
+         * registering a new module in the store named "localePlugin" and sets the given locale,
+         * local locales, and fallback locale.
+         *
+         * @param {Object} store  Vuex store that is going to be used to register the needed
+         * localePlugin module.
+         * @param {Object} localLocales  The local locales information object that should be loaded
+         * and added throught this field. Example:
+         * {
+         *     en_us: { "ripe_twitch.value_a.value_b": "Value Example" },
+         *     pt_pt: { "ripe_twitch.value_a.value_b": "Valor Exemplo"}
+         * }.
+         * @param {String} locale  The ISO-15897 standard locale definition string to be set as the
+         * chosen locale.
+         * @param {String} localeFallback  The ISO-15897 standard locale definition string to be set as the
+         * fallback locale.
+         */
+        setupLocalePlugin(store, localLocales, locale, localeFallback) {

```

Comment:
What is your plan for calling this? In what part of the application?

Reply:
At the start of the app, before any page is loaded